posthog/test/test_filter_model.py

```diff
@@ -19,6 +19,12 @@ def test_old_style_properties(self):
         self.assertEqual(filter.properties[1].operator, None)
         self.assertEqual(filter.properties[1].value, "Mac")
 
+    def test_to_dict(self):
+        filter = Filter(
+            data={"events": [{"id": "$pageview"}], "display": "ActionsLineGraph", "compare": True}
+        ).to_dict()
+        self.assertEqual(list(filter.keys()), ["events", "display", "compare"])
+
```

Comment:
I'd add a test to check that `[]` and `""` values get removed as well, as that seems to be the main change here.

Reply:
Done!